python/tests/test_approximate_nearest_neighbors.py

```diff
@@ -793,3 +808,103 @@ def test_ivfflat_wide_matrix(
     ann_algorithm_test_func(combo=combo, data_shape=data_shape, data_type=data_type)
     duration_sec = time.time() - start
     assert duration_sec < 10 * 60
+
+
+@pytest.mark.parametrize(
+    "algorithm,feature_type",
+    [
+        (
+            "ivfpq",
+            "array",
+        ),
+        (
+            "ivfflat",
+            "vector",
+        ),
+    ],
+)
+@pytest.mark.parametrize("data_type", [np.float32])
+def test_return_fewer_k(
+    algorithm: str,
+    feature_type: str,
+    data_type: np.dtype,
+) -> None:
+    """
+    This tests the corner case where there are less than k neighbors found due to nprobe too small.
+    (1) In ivf_flat and ivf_pq:
```

Comment:
Should this behavior be documented in knn api doc string?

Reply:
Moved it to the class doc string when explaining parameter k.